File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> countMap;
        for ( auto& str : arr) {
            countMap[str]++;
        }
        
        int distinctCount = 0;
        for (const auto& str : arr) {
            if (countMap[str] == 1) {
                distinctCount++;
                if (distinctCount == k) {
                    return str;
                }
            }
        }
        
        return "";
    }
};

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        d={}
        for i in arr:
            d[i]= 1+ d.get(i,0)
        c=0
        for i in d:
            if d[i]==1:
                c+=1
                if c==k:
                    return i
        return ''